### PR TITLE
A-658 Upgrade to use decco

### PR DESCRIPTION
**Pull request by marcesquerra**
Until now, crypto was using the old codecs, not the new refactored 'decco'. If I had used that on OBFT, that would make it had to stick with the old decco, wich, in turn, would have made it imposible to integrate with the new Scalanet.

This provides multicrypto with two versions, using the magic of bazel, one with the old codecs and one with the new decco :)

### CODE REVIEW

**Comment by AlexITC** [comment]
This looks like the whole crypto library, didn't we had this code already? I wonder why I'm not seeing just the changes to match the new decco.

**Comment by marcesquerra** [comment]
@AlexITC For versatility reasons, decco has two packages in it: `io.iohk.codecs` and `io.iohk.decco`. `codecs` has the old style codecs and `decco` has the new refactoring. Thanks to bazel, you can choose to import one or the other, while keeping both in the codebase.

In the same line, this is keeping the pre-existing 'crypto' package that uses the old 'codecs' package and I've cloned the content of 'crypto' into the 'multicrypto' package and then changed everything in that one to use 'decco'. Once everything is stable it's easy to go on and delete the old packages in the libraries.

File: src/io/iohk/multicrypto/encoding/TypedByteString.scala [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+  def decodeFrom(bytes: ByteString): Either[TypedByteStringDecodingError, TypedByteString] = {
+    Codec[TypedByteString].decode(bytes.toByteBuffer) match {
+      case Right(tbs) => Right(tbs)
+      case Left(source) => Left(TypedByteStringDecodingError.DeccoFailedToDecodeTBS(source))
+    }
+  }
+
+  /**
+    * {{{
```
Comment (AlexITC) [inline src/io/iohk/multicrypto/encoding/TypedByteString.scala:64]: This is useless, ideally, we'll have the tests in files.

**Review by AlexITC — APPROVED** [review]
There  is not that much to review, it remembers me to this quote:

> A Pull Request with 10 lines of code leads to 10 issues.
> A Pull Request with 500 lines of code leads to no issues.

**Pull request merged**